plugins/modules/win_domain_computer.py

```diff
@@ -86,6 +89,28 @@
     type: str
     choices: [ absent, present ]
     default: present
+  offline_domain_join:
+    description:
+      - Provisions a computer in the directory and provides a BLOB file that can be used on the target computer/image to join it to the domain while offline.
+      - If the computer already exists, no BLOB will be created/returned, and the module will operate as it would have without offline domain join.
+    type: bool
+  odj_return_blob:
```

Comment:
Instead of having 2 options for this we could combine `offline_domain_join` and `obj_return_blob` into 1 option like `offline_join_blob`. This option will be a string that accepts 3 options

* `none` - (default) no blob is generated/returned
* `output` - Outputs the blob in the module return json
* `path` - Outputs the blob in the path specified by `obj_blob_path`

I'm in no way set on the names here, just suggestions to get the conversation going and to show what I mean.

Reply:
Not completely opposed, but it would remove the possibility of specifying a file to keep and also returning the blob. But I doubt that combination is very useful so this would probably be fine.

Reply:
Went with this, with the slight modification of `odj_blob_path` always being optional and using a temp file if not specified. In `output` mode if a file is specified it will use that, but this mode always deletes the created file. In `path` mode it always leaves the file (even if it's a temp file).